Bottom: c395e3da3639d7a6ca503d0910bcba8af2e597d5
Top:    8c375e2c39f6dd48607ee1775f317194ffa276c7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 23:05:09 +0000

hid/gerber: Fix a couple of typos in drill_sort routine

Should fix sorting of drill holes by coordinate.

(Caught by clang)


---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index c0faf47..820dfdc 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -631,8 +631,8 @@ drill_sort (const void *va, const void *vb)
   if (a->diam != b->diam)
     return a->diam - b->diam;
   if (a->x != b->x)
-    return a->x - a->x;
-  return b->y - b->y;
+    return a->x - b->x;
+  return a->y - b->y;
 }
 
 static int
@@ -955,7 +955,7 @@ use_gc (hidGC gc, int radius)
       if (aptr == NULL)
         pcb_fprintf (stderr, "error: aperture for width %$mS type %s is null\n",
                  linewidth, shape == ROUND ? "ROUND" : "SQUARE");
-      if (f)
+      else if (f)
 	fprintf (f, "G54D%d*", aptr->dCode);
     }
 #if 0
